src/modules/audio/audio.c

```diff
@@ -290,13 +281,14 @@ static void _lovrSourceAssignConverter(Source *source) {
 
   if (!source->converter) {
     ma_data_converter_config config = ma_data_converter_config_init_default();
-    config.formatIn = formats[source->sound->format];
-    config.formatOut = formats[OUTPUT_FORMAT];
+    config.formatIn = miniAudioFormatFromLovr[source->sound->format];
+    config.formatOut = miniAudioFormatFromLovr[OUTPUT_FORMAT];
     config.channelsIn = source->sound->channels;
     config.channelsOut = outputChannelCountForSource(source);
     config.sampleRateIn = source->sound->sampleRate;
     config.sampleRateOut = LOVR_AUDIO_SAMPLE_RATE;
-    arr_expand(&state.converters, 1);
+    // can't use arr_expand because that will destroy the internal pointers inside the converter
```

Comment:
Woops, sorry about that.  Maybe Sources could store the index of their converter?  It would take up less space and avoid having this limit

Reply:
the idea of moving the memory of a struct owned by another library unnerves me :S Something inside of miniaudio could take a pointer to something inside the converter and assume that it wouldn't move between init and deinit, which I think is a sane assumption. If we want to have a dynamic number of converters, I think the choices are either heap allocing each converter, or using a linked list.

Reply:
@bjornbytes I made converters heap allocated instead so the array is an array of pointers and can be safely expanded. I also added converter deiniting, which was missing.